runtime/common/tests/xcm_tests.rs

```diff
@@ -2156,3 +2157,579 @@ fn less_than_min_xcm_fee_should_not_work() {
+        1,
+        X1(Parachain(para_a_id)),
+    )));
+    let para_b_source_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::new(
+        1,
+        X1(Parachain(para_b_id)),
+    )));
+    let units_per_sec = 0;
+
+    let para_a_asset_metadata =
+        create_asset_metadata("ParaAToken", "ParaA", 18, 1, None, false, true);
+    let para_b_asset_metadata =
+        create_asset_metadata("ParaBToken", "ParaB", 18, 1, None, false, true);
+
+    let relay_source_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::parent()));
+    let relay_asset_metadata = create_asset_metadata("Kusama", "KSM", 12, 1, None, false, true);
+
+    // Register ParaA native asset in ParaA
+    let _ = register_assets_on_parachain::<ParaA>(
+        &para_a_source_location,
+        &para_a_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    // Register relaychain native asset in ParaA
+    let relay_asset_id_on_a = register_assets_on_parachain::<ParaA>(
+        &relay_source_location,
+        &relay_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    // register ParaB native asset on ParaA
+    let b_asset_id_on_a = register_assets_on_parachain::<ParaA>(
+        &para_b_source_location,
+        &para_b_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+
+    // register ParaB native asset on ParaB
+    let b_asset_id_on_b = register_assets_on_parachain::<ParaB>(
+        &para_b_source_location,
+        &para_b_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    // register ParaA native asset on ParaB
+    let _ = register_assets_on_parachain::<ParaB>(
+        &para_a_source_location,
+        &para_a_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    // Register relaychain native asset in ParaB
+    let relay_asset_id_on_b = register_assets_on_parachain::<ParaB>(
+        &relay_source_location,
+        &relay_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+
+    let relay_asset_amount_minted_on_a = 10000000;
+    // Initialize some relay chain tokens for alice
+    assert_ok!(ParaA::execute_with(|| {
+        parachain::Assets::mint(
+            parachain::Origin::signed(parachain::AssetManager::account_id()),
+            relay_asset_id_on_a,
+            ALICE,
+            relay_asset_amount_minted_on_a,
+        )
+    }));
+
+    let dest = MultiLocation {
+        parents: 1,
+        interior: X2(
+            Parachain(para_a_id),
+            AccountId32 {
+                network: NetworkId::Any,
+                id: ALICE.into(),
+            },
+        ),
+    };
+
+    // Send some ParaB tokens from Alice on B to Alice on A
+    let amount_to_a = 10000000;
+    let weight = 40;
+    ParaB::execute_with(|| {
+        assert_ok!(parachain::XTokens::transfer(
+            Some(ALICE).into(),
+            parachain::CurrencyId::MantaCurrency(b_asset_id_on_b),
```

Comment:
b was registered but never minted a balance. is the balance transferred here from the genesis endowment to alice on mock net?

Reply:
Yeah this is in the mock setup:
```
pallet_balances::GenesisConfig::<Runtime> {
		balances: vec![(ALICE, INITIAL_BALANCE)],
	}
	.assimilate_storage(&mut t)
	.unwrap();
```
This is kind of hidden logic , which I'm usually against , so I didn't for example add `AssetManager` storage setup there. But this is not too unreasonable to assume, that's why I didn't rewrite it.
Not a big deal if you guys want to make it explicit, just let me know.